docs/proposals/13-automated-seed-management.md

```diff
@@ -0,0 +1,338 @@
+
+**Note:** Even if the controller is called *Horizontal Pod Autoscaler*, it is capable of scaling any resource with a `scale` subresource, using any custom metric. Therefore, initially it was proposed to use this controller directly. However, a number of important drawbacks were identified with this approach, and so it is no longer proposed here.
+
+### SeedAutoscaler Resource
+
+The SeedAutoscaler automatically scales the number of [ManagedSeeds](#managedseeds) in a [ManagedSeedSet](#managedseedsets) based on observed resource utilization. The resource could be any resource that is tracked via the `capacity` and `allocatable` fields in the Seed status, including in particular the number of shoots already scheduled onto the seed.
+
+The SeedAutoscaler is implemented as a custom resource and a new controller. The resource determines the behavior of the controller. The `SeedAutoscaler` resource has a `scaleTargetRef` that specifies the target resource to be scaled, the minimum and maximum number of replicas, as well as a list of metrics. The only supported metric type initially is `Resource` for resources that are tracked via the `capacity` and `allocatable` fields in the Seed status. The resource target can be of type `Utilization` or `AverageValue`.
+
+Example `SeedAutoscaler` resource:
+
+```yaml
+apiVersion: seedmanagement.gardener.cloud/v1alpha1
+kind: SeedAutoscaler
+metadata:
+  name: crazy-botany
+  namespace: garden
+spec:
+  scaleTargetRef:
+    apiVersion: seedmanagement.gardener.cloud/v1alpha1
+    kind: ManagedSeedSet
+    name: crazy-botany
+  minReplicas: 1
+  maxReplicas: 10
+  metrics:
+  - type: Resource # Only Resource is supported
+    resource:
+      name: shoots
+      target:
+        type: Utilization # Utilization or AverageValue
+        averageUtilization: 50
+```
+
+### SeedAutoscaler Controller
+
+`SeedAutoscaler` resources are reconciled by a new *seed autoscaler controller*, either in `gardener-controller-manager` or out-of-tree, similarly to [cluster-autoscaler](https://github.com/gardener/autoscaler). The controller periodically adjusts the number of replicas in a ManagedSeedSet to match the observed average resource utilization to the target specified by user.
+
+**Note:** The SeedAutoscaler controller should perhaps not be limited to evaluating only metrics, it could also take into account also taints, label selectors, etc. This is not yet reflected in the example `SeedAutoscaler` resource below. Such details are intentionally not specified in this GEP, they should be further explored in the issues created to track the actual implementation.
```

Comment:
```suggestion
**Note:** The SeedAutoscaler controller should perhaps not be limited to evaluating only metrics, it could also take into account also taints, label selectors, etc. This is not yet reflected in the example `SeedAutoscaler` resource above. Such details are intentionally not specified in this GEP, they should be further explored in the issues created to track the actual implementation.
```

Reply:
Done